Final 2019 R2 content pushed to open-model-zoo

vmatveic opened the pull request:
Hi All,

This is final PR with latest 2019R2 updates for OMZ demos. After this PR merged all the demos are finally removed from the GitLAB!

Please check it.

@grib0ed0v , please review smart_classroom changes. I've only merged some of our internal changes to GitHUB but no Logic changes are applied (everything is taken from the GitHUB version).

Thanks.

vmatveic commented:
The build was passed internally on Linux with OpenVINO R2. Now I'm going to start our internal CI verification with current OMZ content to verify it.

IRDonch commented on demos/crossroad_camera_demo/README.md:
```diff
@@ -8,7 +8,7 @@ reports person attributes like gender, has hat, has long-sleeved clothes
 a vector of features for each detected person. This vector is used to conclude if it is already detected person or not.
 
-For more information about the pre-trained models, refer to the [model documentation](../../intel_models/index.md).
+For more information about the pre-trained models, refer to the [models documentation](../../intel_models/index.md).
```
This change is not in the internal repository, and it's also questionable WRT grammar, so please revert it.

vmatveic replied:
Fixed

IRDonch commented on demos/crossroad_camera_demo/README.md:
```diff
@@ -31,7 +31,7 @@ compared one-by-one with all previously detected persons vectors using cosine si
 REID value is assigned. Otherwise, the vector is added to a global list, and new REID value is assigned.
 
-> **NOTE**: By default, Open Model Zoo demos expect input with BGR channels order. If you trained your model to work with RGB order, you need to manually rearrange the default channels order in the demo application or reconvert your model using the Model Optimizer tool with `--reverse_input_channels` argument specified. For more information about the argument, refer to **When to Specify Input Sha ...
+> **NOTE**: By default, Open Model Zoo demos expect input with BGR channels order. If you trained your model to work with RGB order, you need to manually rearrange the default channels order in the demo application or reconvert your model using the Model Optimizer tool with `--reverse_input_channels` argument specified. For more information about the argument, refer to **When to Revers Input Chan ...
```
Revers -> Reverse

vmatveic replied:
Done, thanks

Wovchena commented on demos/interactive_face_detection_demo/main.cpp:
```diff
@@ -169,6 +170,9 @@ int main(int argc, char *argv[]) {
         slog::info << "Start inference " << slog::endl;
+        if (!FLAGS_no_show) {
+            std::cout << "Press any key to stop" << std::endl;
+        }
```
This block was intentionally removed by me because there is 
```c++
        std::cout << "To close the application, press 'CTRL+C' here";
        if (!FLAGS_no_show) {
            std::cout << " or switch to the output window and press any key";
        }
````
below. Please, revert.

vmatveic replied:
Agree. Done.

Wovchena commented on demos/security_barrier_camera_demo/README.md:
```diff
@@ -44,8 +44,8 @@ At the end of the sequence, the `VideoFrame` is destroyed and the sequence start
-    API version ............ <version>
-    Build .................. <number>
+        API version ............ <version>
+        Build .................. <number>
```
I seems extra spaces were introduced

vmatveic replied:
yes, moved from GitLAB by mistake. Thanks. Fixed.

Wovchena commented on demos/common/samples/common.hpp:
```diff
@@ -61,7 +58,6 @@ inline std::string &trim(std::string &s) {
     s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
     return s;
 }
-
```
I think we should keep this line :)

Wovchena replied:
And add it to IE samples.

vmatveic replied:
Done :)

Wovchena commented on demos/security_barrier_camera_demo/main.cpp:
```diff
@@ -804,26 +811,26 @@ int main(int argc, char* argv[]) {
+                    std::make_pair(context.platesInfers.getActualInferRequests(), FLAGS_d_lpr)}) {
+                for (InferRequest& ir : net.first) {
+                    ir.Wait(IInferRequest::WaitMode::RESULT_READY);
+                    printPerformanceCounts(ir, std::cout, getFullDeviceName(mapDevices, net.second));
```
Please, revert this change. It is essential to wait for the rest inference calls for the demo when finishing, even if we do not need to `printPerformanceCounts`. It can end up with segmentation fault otherwise.

vmatveic replied:
Yes, got it. Looks like incorrect merge. Fixed.

Wovchena commented on demos/security_barrier_camera_demo/main.cpp:
```diff
@@ -804,26 +811,26 @@ int main(int argc, char* argv[]) {
+                  << "ms / " << std::chrono::seconds(1) / meanOverallTimePerAllInputs << "FPS for " << frameCounter << " frames\n";
+        const double detectionsInfersUsage = static_cast<float>(frameCounter * context.nireq - context.freeDetectionInfersCount)
+            / (frameCounter * context.nireq) * 100;
+        std::cout << "Detection InferRequests usage: " << detectionsInfersUsage << "%\n";
```
Please, revert, because `if (0 != frameCounter) {` avoids division by zero

vmatveic replied:
Done. Incorrect merge again. thanks.

vmatveic commented:
> LGTM, but I'll wait for the other reviewers. Also, I'm going to update the build of OpenVINO used on CI so that the builds can pass.

@IRDonch , thanks ! I working on other comments right now. I will inform you when it's done.

vmatveic commented:
I've fixed all the comments. Thanks guys. Also, I've added last fix from GitLab for `demos/common/samples/console_progress.hpp` file.
The same time I've aligned the content inside of our GitLAB and started internal CI execution. I will inform here when CI is ready.

vmatveic commented:
Jenkins please retry a build

IRDonch commented:
Jenkins please retry a build

IRDonch commented:
Jenkins please retry a build

IRDonch commented:
Jenkins please retry a build

IRDonch commented:
Jenkins please retry a build

vmatveic commented:
All the main internal checks are passed. Some demos test cases are failed. I'm analyzing it now. Inform you here when everything is ready.

vmatveic commented:
I've analyzed the results. Everything is expected, just one check is failed unexpectedly - smart_classroom_demo recognized persons wrongly. Analysis is ongoing.

vmatveic commented:
@IRDonch @Wovchena @kchechil @grib0ed0v @evgeny-izutov @LeonidBeynenson @sovrasov ,
we have analyzed smart_classroom_issue and found that the same problems exists on pure "develop" version of the demo (before my changes). Moreover, I didn't touch this part of source code responsible for reid. Just to repeat - smart_classroom works fine: detects all the persons, recognizes actions but it wrongly recognizes persons (always points to the same person.)  
So, I think in this case we can merge this PR and continue analysis of this smart_classroom_demo issue in scope of a separate case.
Please comment.

vmatveic commented:
@IRDonch @Wovchena @kchechil @grib0ed0v @evgeny-izutov @LeonidBeynenson @sovrasov ,

Thanks to @Wovchena for assistance, we have found the reason of the problem. We need to set `-min_size_fr` option to 73 because small pictures are used within the gallery. So, manual check is passed.

I will update internal CI and re-run smart_classroom_demo verification. After that we are completely ready to merge this PR.
I will inform you about verification results asap.

vmatveic commented:
@IRDonch @kchechil @snosov1 @Wovchena according to current execution log the CI results are better, all the faces/actions are detected and recognized. Just reference values must be recollected. We will handle it during the CI migration to support OMZ demos from GitHUB. So, I hink this PR can be merged. Any objections?
